app/code/Magento/Customer/Block/Form/Register.php

```diff
@@ -214,4 +214,23 @@ public function getRequiredCharacterClassesNumber()
     {
         return $this->_scopeConfig->getValue(AccountManagement::XML_PATH_REQUIRED_CHARACTER_CLASSES_NUMBER);
     }
+    
+    /**
+     * Address fields availability
+     */
+    public function getShowAddressFields()
```

Comment:
1)  Due to [Magento backward-compatible guide](http://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development/) we can't create new public methods in classes that are marked as API.
2) The method returns a boolean value, so this method should have the name with `is` prefix instead of 'get'
3) It's a strange approach to create 3 blocks inside the block just for check status and with this approach, also it's hard dependencies to that blocks and we willn't have ability to replace their.

In general I think we should find the reason why `showAddressFields` don't set correct and fix it.